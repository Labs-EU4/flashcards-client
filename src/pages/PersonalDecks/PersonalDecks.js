import React, {useEffect} from "react";
import List from "../../components/ListDeckInfo/DeckList";
import Header from "../../components/ListDeckInfo/HeaderSearchBar";
import styles from "./PersonalDecks.module.css";
import Dashboard from "../../layout/Dashboard/Dashboard";
import {getAllPersonalDecks} from "../../state/actions/decks";
import {connect} from "react-redux";

export function TestPersonalDecks({getAllPersonalDecks, decks}) {
  useEffect(() => {
    getAllPersonalDecks();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div className={styles.personal}>
      <Header page={"Deck Library"} />
      <List decks={decks} />
    </div>
  );
}

export function PersonalDecks({getAllPersonalDecks, decks}) {
  useEffect(() => {
    getAllPersonalDecks();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Dashboard>
      <div className={styles.personal}>
        <Header page={"Deck Library"} />
        <List decks={decks} page={"personal"} />
      </div>
    </Dashboard>
  );
}

function mapStateToProps(state) {
  return {
    decks: state.deckState.personalDeckState,
  };
}

export default connect(mapStateToProps, {getAllPersonalDecks})(PersonalDecks);
