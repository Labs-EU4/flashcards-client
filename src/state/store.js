import thunk from "redux-thunk";
import {combineReducers, createStore, applyMiddleware, compose} from "redux";
import authReducer from "./reducers/auth";
import {
  currentDeckReducer,
  publicDecksReducer,
  personalDecksReducer,
  deckReducer,
} from "./reducers/decks";

const mainReducer = combineReducers({
  authState: authReducer,
  publicDeckState: publicDecksReducer,
  personalDeckState: personalDecksReducer,
  currentDeckState: currentDeckReducer,
  deckState: deckReducer,
});

const devTools =
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    ? window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    : n => n;

const store = createStore(
  mainReducer,
  {},
  compose(applyMiddleware(thunk /* ,etc , other middlewares */), devTools)
);

export default store;
