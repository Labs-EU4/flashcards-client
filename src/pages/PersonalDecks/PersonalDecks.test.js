import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import {TestPersonalDecks, PersonalDecks} from "./PersonalDecks";
import {Provider} from "react-redux";
import store from "../../state/store";
import {BrowserRouter} from "react-router-dom";

const mockGetDecks = jest.fn(() => true);

const PageHeader = () => {
  return wrapper.queryByText("Deck Library");
};

const NoDecksMessage = () => {
  return wrapper.queryByText("You have no decks right now");
};

let wrapper;
beforeEach(() => {
  wrapper = rtl.render(
    <TestPersonalDecks getAllPersonalDecks={mockGetDecks} decks={[]} />
  );
});

it("renders without crashing", () => {
  expect(wrapper.container).toMatchSnapshot();
});

it("displays the right text", () => {
  expect(PageHeader()).toBeInTheDocument();
  expect(NoDecksMessage()).toBeInTheDocument();
});
