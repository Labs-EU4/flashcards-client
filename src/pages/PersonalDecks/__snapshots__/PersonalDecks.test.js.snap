// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="personal"
  >
    <div
      class="searchContainer"
    >
      <div
        class="cover"
      >
        <h1
          class="heading"
          data-testid="page"
        >
          Deck Library
        </h1>
        <span
          class="ant-input-search search ant-input-affix-wrapper"
        >
          <input
            class="ant-input"
            placeholder="Search"
            type="text"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <i
              aria-label="icon: search"
              class="anticon anticon-search ant-input-search-icon"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </i>
          </span>
        </span>
        <div
          class="sort"
        >
          <p>
            Sort by:
          </p>
          <div
            class="sortselect ant-select ant-select-enabled"
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-select-selection
            ant-select-selection--single"
              role="combobox"
              tabindex="0"
            >
              <div
                class="ant-select-selection__rendered"
              >
                <div
                  class="ant-select-selection-selected-value"
                  style="display: block; opacity: 1;"
                  title="Option 1"
                >
                  Option 1
                </div>
              </div>
              <span
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <i
                  aria-label="icon: down"
                  class="anticon anticon-down ant-select-arrow-icon"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main_content"
    >
      <h2
        class="noName"
      >
        You have no decks right now
      </h2>
    </div>
  </div>
</div>
`;
